controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -335,13 +335,13 @@ private SegmentRange convert(final String scope,
     }
 
     public CompletableFuture<TxnStatus> commitTransaction(final String scope, final String stream, final TxnId
-            txnId) {
+            txnId, String writerId, long time) {
```

Comment:
We should be very explicit on the definition of this time parameter. I've added these same two fields to `commitTransaction` in the watermarking branch, with a different goal. Are these going to be compatible? If so we should make explicit that this is a application defined time.
Also:
Nit: Seems like writerId should go before TxnId to me

Reply:
I do not need these fields for transaction ordering.. 
i was updating the transaction metadata (ActiveTxRecord) so i thought i will update it once with all the fields that we know we will need for watermarking. 
So they represent the same thing as has been discussed in watermarking PDP.